Source: upstream developer Guy Harris
https://www.wireshark.org/lists/wireshark-users/200806/msg00206.html
Reason: avoid messages: arptype 65534 not supported by libpcap - falling back to cooked socket.

Index: pcap-linux.c
===================================================================
RCS file: /tcpdump/master/libpcap/pcap-linux.c,v
retrieving revision 1.148
diff -c -r1.148 pcap-linux.c
*** pcap-linux.c	14 Apr 2008 21:04:51 -0000	1.148
--- pcap-linux.c	24 Jun 2008 06:43:49 -0000
***************
*** 1539,1544 ****
--- 1539,1555 ----
  		handle->linktype = DLT_LINUX_LAPD;
  		break;
  
+ #ifndef ARPHRD_NONE
+ #define ARPHRD_NONE	0xFFFE
+ #endif
+ 	case ARPHRD_NONE:
+ 		/*
+ 		 * No link-layer header; packets are just IP
+ 		 * packets, so use DLT_RAW.
+ 		 */
+ 		handle->linktype = DLT_RAW;
+ 		break;
+ 
  	default:
  		handle->linktype = -1;
  		break;
